crates/iota-light-client/src/main.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{path::PathBuf, str::FromStr};
+
+use anyhow::Context;
+use clap::{Parser, Subcommand};
+use iota_light_client::{
+    checkpoint::sync_and_check_checkpoints,
+    config::Config,
+    package_store::RemotePackageStore,
+    verifier::{get_verified_effects_and_events, get_verified_object},
+};
+use iota_package_resolver::Resolver;
+use iota_types::{
+    base_types::ObjectID,
+    digests::TransactionDigest,
+    object::{Data, bounded_visitor::BoundedVisitor},
+};
+use tracing::debug;
+
+// Define the `GIT_REVISION` and `VERSION` consts
+bin_version::bin_version!();
+
+#[derive(Parser, Debug)]
+#[command(
+    name = env!("CARGO_BIN_NAME"),
+    about = env!("CARGO_PKG_DESCRIPTION"),
+    author,
+    version = VERSION,
+    propagate_version = true,
+)]
+struct Args {
+    /// Sets a custom config file
+    #[arg(short, long, value_name = "PATH")]
+    config: Option<PathBuf>,
+    #[command(subcommand)]
+    command: LightClientCommand,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum LightClientCommand {
+    /// Sync light client
+    Sync,
+    /// Check a transaction for inclusion
+    CheckTransaction {
+        /// Transaction digest
+        #[arg(value_name = "BASE58")]
+        transaction_digest: String,
+    },
+    /// Check an object for inclusion
+    CheckObject {
+        /// Object ID
+        #[arg(value_name = "HEX")]
+        object_id: String,
```

Comment:
Why not use `ObjectID`?

Reply:
Honestly I have just gone with upstream here, but for sure I can change as it's probably not worth a separate PR.

Reply:
https://github.com/iotaledger/iota/pull/6135/commits/a2eaa161fb5fa26cefe3c62dae14d3ba6bc9e135